std/meta.d

```diff
@@ -1309,6 +1309,52 @@ private template staticMerge(alias cmp, int half, Seq...)
     }
 }
 
+/**
+ * Allows subsetting of multiple members of tuple types that are not
+ * adjacent to each other, i.e. cannot be sliced.
+ *
+ * Evaluates to $(D AliasSeq!(T[range[0]], T[range[1]], ..., T[range[n]])).
+ */
+
+template staticSubset(alias range, T...)
+{
+	import std.range : isInputRange;
+	import std.traits : isArray, isNarrowString;
+
+	alias ArrT = typeof(range);
+	static if (isArray!ArrT && !isNarrowString!ArrT)
+	{
+		static if(range.length == 0)
+		{
+			alias staticSubset = AliasSeq!();
+		} else {
+			alias staticSubset = AliasSeq!(T[range[0]], staticSubset!(range[1 .. $], T));
```

Comment:
This will fail badly if the element type of `range` is not integral - this can be handled nicely by a template constraint, such as `isInputRange!(typeof(range)) && is(ElementType!(typeof(range)) : size_t)`